include/gpioMonitor.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <sdbusplus/asio/connection.hpp>
+
+namespace vpd
+{
+/**
+ * @brief class for GPIO event handling.
+ *
+ * Responsible for detecting events and handling them. It continuously
+ * monitors the presence of the FRU. If any attachment or detachment is
+ * detected, it enables or disables the FRU's output GPIO and binds or
+ * unbinds the driver accordingly.
+ */
+class GpioEventHandler
+{
+  public:
+    GpioEventHandler() = delete;
+    ~GpioEventHandler() = delete;
+    GpioEventHandler(const GpioEventHandler&) = delete;
+    GpioEventHandler& operator=(const GpioEventHandler&) = delete;
+    GpioEventHandler(GpioEventHandler&&) = delete;
+    GpioEventHandler& operator=(GpioEventHandler&&) = delete;
+
+    /**
+     * @brief Constructor
+     *
+     * @param[in] i_presPin - GPIO presence pin
```

Comment:
Parameters mentioned here does not matches the parameters being passed actually.

Reply:
Updated